packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.tsx

```diff
@@ -110,6 +112,17 @@ export class UnconnectedSearchResults extends React.PureComponent<SearchResultsP
     history.push(getUrl({ ...urlState, view }));
   };
 
+  onDownloadResultsClicked = () => {
+    const file = new Blob([JSON.stringify(this.props.tracesToDownload)], {type : "application/json"});
+    const element = document.createElement("a");
+    element.href = URL.createObjectURL(file);
+    element.download = "traces-" + Date.now() + ".json";
+    document.body.appendChild(element);
+    element.click();
+    URL.revokeObjectURL(element.href)
+    element.remove()
```

Comment:
Reading this as not a web developer, is this the preferred way of executing downloads from a page? Do you have some pointer to documentation describing it?

Reply:
I am not primarily a web developer. But when I was looking for the best way of executing downloads from a page I found bunch of similar solution how you can do it.  Here some references:

- https://www.techiediaries.com/javascript-blobs-object-urls/ (Date: 02 Oct 2022)
- https://theroadtoenterprise.com/blog/how-to-download-csv-and-json-files-in-react (27 Aug, 2021)

element.click() --> it's simulate a click 
URL.revokeObjectURL(element.href) --> "it is recommended to release object URLs to optimize performance and reduce memory consumption."  In the first article it is good explained.